src/test/scala/cromwell/engine/backend/CromwellRuntimeAttributeSpec.scala

```diff
@@ -341,241 +351,245 @@ object CromwellRuntimeAttributeSpec {
+    }
     ex.getMessage should include ("is an invalid memory unit")
   }
 
-  "WDL file with an invalid local disk" should "say so" in {
+  it should "reject a task with an invalid 'disks' parameter" in {
     val ex = intercept[IllegalArgumentException] {
-      NamespaceWithWorkflow.load(WorkflowWithMessedUpLocalDisk).toCromwellRuntimeAttributes(BackendType.JES)
+      runtimeAttributes(WorkflowWithMessedUpLocalDisk, "messed_up_disk", jesBackend)
     }
-
     ex.getMessage should include(
-      "'Disk1 123 LOCAL' should be in form 'NAME SIZE TYPE', with SIZE blank for LOCAL, otherwise SIZE in GB")
+      "'Disk1 123 LOCAL' should be in form 'NAME SIZE TYPE', with SIZE blank for LOCAL, otherwise SIZE in GB"
+    )
   }
 
-  "WDL file with an invalid disk size" should "say so" in {
+  it should "reject a task with an invalid 'disks' parameter (2)" in {
     val ex = intercept[IllegalArgumentException] {
-      NamespaceWithWorkflow.load(WorkflowWithMessedUpDiskSize).toCromwellRuntimeAttributes(BackendType.JES)
+      runtimeAttributes(WorkflowWithMessedUpDiskSize, "messed_up_disk", jesBackend)
     }
-
     ex.getMessage should include("123.0 not convertible to a Long")
   }
 
-  "WDL file with an invalid disk type" should "say so" in {
+  it should "reject a task with an invalid 'disks' parameter (3)" in {
     val ex = intercept[IllegalArgumentException] {
-      NamespaceWithWorkflow.load(WorkflowWithMessedUpDiskType).toCromwellRuntimeAttributes(BackendType.JES)
+      runtimeAttributes(WorkflowWithMessedUpDiskType, "messed_up_disk", jesBackend)
     }
-
     ex.getMessage should include("Disk TYPE SDD should be one of LOCAL, SSD, HDD")
   }
 
-  "Default RuntimeAttributes" should "be accessible for all tasks" in {
-    val namespaceWithoutRuntime = NamespaceWithWorkflow.load(ManyTaskWorkflowWithoutRuntime)
-    WorkflowOptions.fromJsonString(DefaultRuntimeAttributesOptionsFile) match {
-      case Success(options) => namespaceWithoutRuntime.toCromwellRuntimeAttributes(options) foreach { attributes =>
-        assert(attributes.docker.isDefined)
-        assert(attributes.docker.get.equals(DefaultRuntimeAttributesOptionsDockerValue))
-      }
-      case Failure(e: Throwable) => fail(s"Couldn't parse $DefaultRuntimeAttributesOptionsFile as JSON.", e)
-      case _ => fail(s"Couldn't parse $DefaultRuntimeAttributesOptionsFile as JSON.")
-    }
+  it should "allow runtime attributes to be expressions that reference task inputs" in {
+    val attributes = runtimeAttributes(WorkflowWithRuntimeAttributeExpressions, "test", jesBackend)
+    attributes.memoryGB shouldEqual 7
+    attributes.disks shouldEqual Vector(
+      new Disk().setName("Disk1").setSizeGb(9L).setType("PERSISTENT_SSD").setAutoDelete(true),
+      CromwellRuntimeAttributes.LocalizationDisk.toDisk
+    )
+    attributes.docker shouldEqual Some("ubuntu:latest")
   }
 
-  "Default RuntimeAttributes" should "NOT override WDL values" in {
```

Comment:
Though I suppose your test is different because it asserts it for *all* tasks.  I can alter the test below to test for all tasks in a WDL file instead of just one.